### Eclipse Workspace Patch 1.0
#P org.eclipse.php.core
Index: src/org/eclipse/php/core/index/PhpIndexingVisitorExtension.java
===================================================================
RCS file: /cvsroot/tools/org.eclipse.pdt/plugins/org.eclipse.php.core/src/org/eclipse/php/core/index/PhpIndexingVisitorExtension.java,v
retrieving revision 1.1
diff -u -r1.1 PhpIndexingVisitorExtension.java
--- src/org/eclipse/php/core/index/PhpIndexingVisitorExtension.java	30 Jul 2009 09:12:20 -0000	1.1
+++ src/org/eclipse/php/core/index/PhpIndexingVisitorExtension.java	25 Jun 2011 17:56:31 -0000
@@ -13,6 +13,7 @@
 
 import org.eclipse.dltk.ast.ASTNode;
 import org.eclipse.dltk.ast.ASTVisitor;
+import org.eclipse.dltk.core.index2.IIndexingRequestor;
 import org.eclipse.dltk.core.index2.IIndexingRequestor.DeclarationInfo;
 import org.eclipse.dltk.core.index2.IIndexingRequestor.ReferenceInfo;
 
@@ -24,6 +25,8 @@
  */
 public abstract class PhpIndexingVisitorExtension extends ASTVisitor {
 
+	protected IIndexingRequestor requestor;
+
 	/**
 	 * This is a last chance before modifying element declaration information
 	 * before inserting it into index.
@@ -47,4 +50,9 @@
 	 */
 	public void modifyReference(ASTNode node, ReferenceInfo info) {
 	}
+
+	public void setRequestor(IIndexingRequestor requestor) {
+
+		this.requestor = requestor;
+	}
 }
Index: src/org/eclipse/php/internal/core/index/PhpIndexingVisitor.java
===================================================================
RCS file: /cvsroot/tools/org.eclipse.pdt/plugins/org.eclipse.php.core/src/org/eclipse/php/internal/core/index/PhpIndexingVisitor.java,v
retrieving revision 1.13
diff -u -r1.13 PhpIndexingVisitor.java
--- src/org/eclipse/php/internal/core/index/PhpIndexingVisitor.java	24 Mar 2011 01:12:02 -0000	1.13
+++ src/org/eclipse/php/internal/core/index/PhpIndexingVisitor.java	25 Jun 2011 17:56:32 -0000
@@ -99,8 +99,11 @@
 				extensionElements.length);
 		for (IConfigurationElement element : extensionElements) {
 			try {
-				extensions.add((PhpIndexingVisitorExtension) element
-						.createExecutableExtension(CLASS_ATTR));
+
+				PhpIndexingVisitorExtension extension = (PhpIndexingVisitorExtension) element
+						.createExecutableExtension(CLASS_ATTR);
+				extension.setRequestor(requestor);
+				extensions.add(extension);
 			} catch (CoreException e) {
 				Logger.logException(e);
 			}
